src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from 'react';
+
+import { Button } from 'components/Button/Button';
+
+import { useIndex } from 'hooks/useIndex';
+
+import styles from './Carousel.module.css';
+
+type Props = {
+  items: JSX.Element[];
+  className?: string;
+};
+
+export default function Carousel({ items, className }: Props) {
+  const { index, next, prev, jumpTo, setMaxIndex } = useIndex({
+    initial: 0,
+  });
+
+  useEffect(() => {
+    setMaxIndex(items.length - 1);
+  }, [items]);
+
+  const carouselItems = items.map((item, itemIndex) => (
+    <li
+      key={`carouselItem-${itemIndex}`}
+      className={styles['item-container']}
+    >
+      {item}
+    </li>
+  ));
+
+  const nextItemHandler = () => {
+    if (index === carouselItems.length - 1) {
```

Comment:
next() returns true if it was possible to call.
Just as a note, this could have gone something like
`if (!next()) jumpTo(0)`